modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/connectivity/oauth/authcode/AuthorizationCodeConnectionProviderWrapper.java

```diff
@@ -68,13 +71,15 @@ public C connect() throws ConnectionException {
   private void updateAuthState() {
     final Object delegate = getDelegateForInjection();
     ResourceOwnerOAuthContext context = getContext();
+    boolean forceInvalidateStatusRetrieval = !muleContext.getClusterId().isEmpty();
```

Comment:
determine this outside of oauth logic, do not depend on the mule?Context or cluster specific stuff